DSharpPlus/Entities/DiscordGuild.cs

```diff
@@ -1710,6 +1710,36 @@ public override int GetHashCode()
             return this.Id.GetHashCode();
         }
 
+        /// <summary>
+        /// Compares the inputted file size to the file limit of the guild.
+        /// <para>Returns whether the size breaks this limit and also gives the out variable of the guild's file limit, in megabytes.</para>
+        /// </summary>
+        internal bool CheckGuildFileSize(long bytes, out int limit)
+        {
+            var sizeInMb = (bytes / 1024) / 1024;
+
+            switch (this.PremiumTier)
+            {
+                case PremiumTier.None:
+                    limit = 8;
+                    break;
+
+                case PremiumTier.Tier_2:
+                    limit = 50;
+                    break;
+
+                case PremiumTier.Tier_3:
+                    limit = 100;
+                    break;
+
+                default:
+                    limit = 8;
+                    break;
+            }
+            
+            return sizeInMb < limit;
+        }
+
```

Comment:
* You should multiply the limit by 1048576 or 1024*1024 or whatever, instead of dividing
* The less-than operator should be less-than-equals
* Same thing for the other method

Reply:
Should be good now.